source/simulator/core.cc

```diff
@@ -1043,6 +1046,7 @@ namespace aspect
             {
               coupling[x.velocities[d]][x.pressure] = DoFTools::always;
               coupling[x.pressure][x.velocities[d]] = DoFTools::always;
+              coupling[x.pressure][x.pressure] = DoFTools::always;
```

Comment:
I think the reason is that your signal only modifies ``current_constraints`` but you also need to modify ``constraints``, which is the ``ConstraintMatrix`` used for construction the sparsity pattern of the matrix. i think ``signals.pre_compute_no_normal_flux_constraints`` should work.